src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -522,9 +522,7 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
     protected function denormalizeParameter(\ReflectionClass $class, \ReflectionParameter $parameter, $parameterName, $parameterData, array $context, $format = null)
     {
         try {
-            if (\PHP_VERSION_ID < 70100 && null !== $parameterClass = $parameter->getClass()) {
-                $parameterClass = $parameterClass->name;
-            } elseif (\PHP_VERSION_ID >= 70100 && $parameter->hasType() && ($parameterType = $parameter->getType()) && !$parameterType->isBuiltin()) {
+            if ($parameter->hasType() && ($parameterType = $parameter->getType()) && !$parameterType->isBuiltin()) {
```

Comment:
we can remove the call to hasType, isn't it?

Reply:
That is correct and we could do this on 3.4 already.

Reply:
done in ca695e55e8

Reply:
and 236a2145fc for 4.4